hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -276,6 +276,8 @@ private void refreshTimeline() throws IOException {
       // to generic records for writing
       InputBatch<Dataset<Row>> dataAndCheckpoint =
           formatAdapter.fetchNewDataInRowFormat(resumeCheckpointStr, cfg.sourceLimit);
+      SchemaProvider schemaProviderFromFetched = dataAndCheckpoint.getBatch().isPresent()
+          ? dataAndCheckpoint.getSchemaProvider() : null;
```

Comment:
when fetching in row format, this change should not be needed since RowBasedSchemaProvider is already getting initialised at the end.

Reply:
@xushiyan Here this change should not be needed. WDYT?